Make sure order of analyses is saved properly.

JorisGoosen opened the pull request:
- Fixes https://github.com/jasp-stats/jasp-issues/issues/639
- Also moved all the functions for moving analyses together and into Analyses, Im not sure why part of it was on MainWindow.

TimKDJ commented:
That issue goes nowhere for me

JorisGoosen commented:
@TimKDJ what do you mean by that statement?

TimKDJ commented:
The issue link https://github.com/jasp-stats/jasp-issues/issues/639o doesn't link to an issue for me

JorisGoosen commented:
Haha, Im not used to this mac keyboard, the fn is on the place of ctrl which means fn+o in nano gives superfluous 'o' :-p 

Ill fix it and push-force it.

JorisGoosen committed:
Make sure order of analyses is saved properly.
- Fixes https://github.com/jasp-stats/jasp-issues/issues/639
- Also moved all the functions for moving analyses together and into Analyses, Im not sure why part of it was on MainWindow.

JorisGoosen commented:
So @boutinb how do you like this PR?

boutinb reviewed (commented):
By reviewing your code, I still don't see why it was not storing the new order in the JASP file?

JorisGoosen commented on JASP-Desktop/analysis/analyses.cpp:
```diff
@@ -248,8 +248,8 @@ Json::Value Analyses::asJson() const
 
-	for(auto idAnalysis : _analysisMap)
-			analysesDataList.append(idAnalysis.second->asJSON());
+	applyToAll([&analysesDataList](const Analysis * analysis)
```
@boutinb here is the actual fix, the rest is just rearranging the deckchairs on the titanic. 
But the problem was that using an iterator on a std::map does not give the proper order at all. So the analyses weren't stored in the right order in the jasp-file. Which is why order was lost.

The function applyToAll was specifically added to make sure all analyses are processed in the right order. Why we didn't do it here, I don't know.
